iree/samples/static_library/README.md

```diff
@@ -0,0 +1,60 @@
+# IREE "Static Library" sample
+
+This sample shows how to:
+1. Produce a static library and bytecode module with IREE's compiler
+2. Compile the static library into a program using the `static_library_loader`
+3. Run the demo with the module using functions exported by the static library
+
+The model compiled into the static library exports a single function `simple_mul` that returns the multiplication of two tensors:
+
+```mlir
+func @simple_mul(%arg0: tensor<4xf32>, %arg1: tensor<4xf32>) -> tensor<4xf32>
+    attributes { iree.module.export } {
+  %0 = "mhlo.multiply"(%arg0, %arg1) {name = "mul.1"} : (tensor<4xf32>, tensor<4xf32>) -> tensor<4xf32>
+  return %0 : tensor<4xf32>
+}
+```
+
+## Background
+
+IREE's `static_library_loader` allows applications to inject a set of static libraries that can be resolved at runtime by name. This can be particularly useful on "bare metal" or embedded systems running IREE that lack operating systems or the ability to load shared libraries in binaries.
+
+When static library output is enabled, `iree-translate` produces a separate static library to compile into the target program. At runtime bytecode module instructs the VM which static libraries to load exported functions from the model.
+
+## Instructions
+Note: run these commands from IREE's github root.
+
+1. Configure CMake with the `IREE_BUILD_STATIC_LIBRARY_SAMPLES` then build the `iree_translate` CMake target (see
+   [here](https://google.github.io/iree/building-from-source/getting-started/)
+   for general instructions on building using CMake):
+
+   ```shell
+   cmake -B ../iree-build/ -IREE_BUILD_STATIC_LIBRARY_SAMPLES=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo .
+   cmake --build ../iree-build/ --target iree_tools_iree-translate
```

Comment:
if you're doing this then you really can just key off of IREE_BUILD_COMPILER + IREE_BUILD_SAMPLES - as unless IREE_BUILD_COMPILER is set then you won't be able to build iree-translate